app/ide-desktop/lib/content/esbuild-config.ts

```diff
@@ -106,27 +78,82 @@ export async function* filesToCopyProvider(wasmArtifacts: string, assetsPath: st
  * Generate the builder options.
  */
 export function bundlerOptions(args: Arguments) {
-    const { outputPath, ensoglAppPath, wasmArtifacts, assetsPath } = args
+    const { outputPath, ensoglAppPath, wasmArtifacts, assetsPath, devMode } = args
     const buildOptions = {
         // Disabling naming convention because these are third-party options.
         /* eslint-disable @typescript-eslint/naming-convention */
         absWorkingDir: THIS_PATH,
         bundle: true,
-        entryPoints: [path.resolve(THIS_PATH, 'src', 'index.ts')],
+        loader: {
+            '.html': 'copy',
+            '.css': 'copy',
+            '.wasm': 'copy',
+            '.svg': 'copy',
+            '.png': 'copy',
+            '.ttf': 'copy',
+        },
+        entryPoints: [
+            pathModule.resolve(THIS_PATH, 'src', 'index.ts'),
+            pathModule.resolve(THIS_PATH, 'src', 'index.html'),
+            pathModule.resolve(THIS_PATH, 'src', 'run.js'),
+            pathModule.resolve(THIS_PATH, 'src', 'style.css'),
+            pathModule.resolve(THIS_PATH, 'src', 'docsStyle.css'),
+            ...wasmArtifacts.split(pathModule.delimiter),
+            ...fsSync
+                .readdirSync(assetsPath)
+                .map(fileName => pathModule.resolve(assetsPath, fileName)),
+        ].map(path => ({ in: path, out: pathModule.basename(path, pathModule.extname(path)) })),
         outdir: outputPath,
         outbase: 'src',
         plugins: [
+            {
+                // This is a workaround that is needed because esbuild panics when using `loader: { '.js': 'copy' }`.
+                name: 'pkg-js-is-cjs',
+                setup: build => {
+                    build.onLoad({ filter: /\/pkg.js$/ }, async ({ path }) => ({
+                        contents: await fs.readFile(path),
+                        loader: 'copy',
+                    }))
+                },
+            },
+            {
+                name: 'copy-directories',
+                setup: build => {
+                    build.onResolve({ filter: /[/\\][^./\\]+$/ }, ({ path, kind }) =>
+                        kind === 'entry-point'
+                            ? { path, namespace: 'copy-directories', watchDirs: [path] }
```

Comment:
I don't think it will update when the files content change, `watchDir` is only for the entry list changes. Also, it does not watch subdirectories.

Reply:
this is true, however afaict the same is true for the old copy plugin - since directories are copied via `fs.cp(from, to, { recursive: true })` and then pushed to `watchFiles`. since these are resolved on every build (i think) because they are entry points, they are probably always copied - i can add a comment after extracting out to a plugin

Reply:
wait actually... it still won't update when file contents change. we might need to use `chokidar` and manually trigger rebuilds? alternatively leave as-is, but with a comment noting this as a weakness of this approach

Reply:
oh, alternatively `chokidar` can be used to continuously keep the directories in sync after the initial `cp`, skipping all `onResolve` and `onLoad` except for the first `onResolve`, which would register the watcher